File: vaadin-7-cookbook/Chapter 1/01_04_vaadin-in-scala/src/main/scala/app/MyVaadinUI.scala
package app

import com.vaadin.ui._
import com.vaadin.server.VaadinRequest

class MyVaadinUI extends UI {
	def init(request: VaadinRequest) = {
		val layout = new VerticalLayout()
        layout.setMargin(true)
        setContent(layout)
		layout.addComponent(new Label("Hello Vaadin user."))
	}
}

File: vaadin-7-cookbook/Chapter 6/06_02_system-messages/src/main/java/com/app/MyVaadinUI.java
package com.app;

import com.vaadin.annotations.PreserveOnRefresh;
import com.vaadin.annotations.Theme;
import com.vaadin.server.*;
import com.vaadin.ui.*;
import com.vaadin.ui.Button.ClickEvent;

/**
 * The Application's "main" class
 */
@Theme("mytheme")
public class MyVaadinUI extends UI {

    @Override
    protected void init(VaadinRequest request) {
        final VerticalLayout layout = new VerticalLayout();
        layout.setMargin(true);
        setContent(layout);

        Button btn = new Button("Click Button");
        btn.addClickListener(new Button.ClickListener() {
            @Override
            public void buttonClick(ClickEvent clickEvent) {
                Label lbl = new Label("Turn off the server and click the button again.");
                layout.addComponent(lbl);
            }
        });
        layout.addComponent(btn);
    }
}
